scheduler/pkg/klcpermit/workflow_manager.go

```diff
@@ -54,73 +56,95 @@ type Manager interface {
 type WorkloadManager struct {
 	dynamicClient dynamic.Interface
 	Tracer        trace.Tracer
+	bindCRDSpan   map[string]trace.Span
 }
 
 func NewWorkloadManager(d dynamic.Interface) *WorkloadManager {
 	sMgr := &WorkloadManager{
 		dynamicClient: d,
 		Tracer:        otel.Tracer("keptn/scheduler"),
+		bindCRDSpan:   make(map[string]trace.Span, 100),
 	}
 	return sMgr
 }
 
-var bindCRDSpan = make(map[string]trace.Span, 100)
+func (sMgr *WorkloadManager) ObserveWorkloadForPod(ctx context.Context, handler framework.WaitingPod, pod *corev1.Pod) {
+	factory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(sMgr.dynamicClient, 0, pod.GetNamespace(), nil)
```

Comment:
maybe it is worth to test if this filters works instead of everytime checking for it in the handler
```
workloadInstanceName := getCRDName(pod)
labelOptions :=func(opts *v1.ListOptions) {
		opts.FieldSelector = "metadata.name=" + workloadInstanceName
	}
factory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(sMgr.dynamicClient, 0, pod.GetNamespace(), labelOptions)
```

Reply:
I was thinking about this as well, but we currently do not have such labels in WI nor anywhere, do we want to introduce them? If yes, we should introduce them to all resources, not just for WI. But I guess that should be part of a separate PR, which should be merged before this one

Reply:
yeah probably worth writing a ticket for this one or discuss it :)